tools/tsp-client/src/commands.ts

```diff
@@ -474,13 +474,65 @@ export async function generateConfigFilesCommand(argv: any) {
   if (Object.keys(overrideJson).length > 0) {
     emitterPackageJson["overrides"] = overrideJson;
   }
-  
-  const emitterPath = resolveEmitterPathFromArgs(argv) ?? joinPaths(await getRepoRoot(outputDir), defaultRelativeEmitterPackageJsonPath);
-  
-  await writeFile(
-    emitterPath,
-    JSON.stringify(emitterPackageJson, null, 2),
-  );
+
+  const emitterPath =
+    resolveEmitterPathFromArgs(argv) ??
+    joinPaths(await getRepoRoot(outputDir), defaultRelativeEmitterPackageJsonPath);
+
+  let existingEmitterPackageJson: Record<string, any> | undefined;
+  try {
+    existingEmitterPackageJson = JSON.parse(await readFile(emitterPath, "utf8"));
+  } catch (err) {
+    Logger.debug(
+      `Couldn't read ${basename(emitterPath)}. If the file exists it will be over-written. Error: ${err}`,
+    );
+  }
+  // If there's an existing emitter-package.json, we need to check for any manually added devDependencies
+  if (existingEmitterPackageJson) {
+    // Register all manually pinned dependencies and their current values
+    const manualDevDependencies = {};
+    for (const [key, value] of Object.entries(
+      existingEmitterPackageJson["devDependencies"] ?? {},
+    )) {
+      if (!Object.keys(emitterPackageJson["devDependencies"] ?? {}).includes(key)) {
+        Object.assign(manualDevDependencies, { [key]: value });
+      }
+    }
+
+    // Attempt to read package-lock.json and find the version of the manually added dependencies
```

Comment:
Why do we need the version from the package-lock? Shouldn't we use the version from emitter package JSON file?

Reply:
Let me know and I'll update accordingly